include/Zycore/Defines.h

```diff
@@ -64,6 +64,9 @@
 #if defined(__clang__)
 #   define ZYAN_CLANG
 #   define ZYAN_GNUC
+#   if defined(_MSC_VER)
+#       define ZYAN_CLANG_CL
+#   endif
```

Comment:
Hmm. Wondering whether `ZYAN_GNUC` should go into `else` branch of this new define? I'm guessing that clang-cl won't accept GNU extensions, but I'm not sure.

Reply:
iirc, although in `-fms-compatibility` mode, clang still supports many gnu extensions, so imo the GNUC macro should be defined.

Reply:
e.g. `__attribute__((cleanup))`

Reply:
Ok, makes sense then -- thanks for clarifying!